pkg/vmnetworkscontroller/vm_controller.go

```diff
@@ -0,0 +1,99 @@
+package vmnetworkscontroller
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/kubevirt/ipam-extensions/pkg/claims"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	virtv1 "kubevirt.io/api/core/v1"
+)
+
+// VirtualMachineReconciler reconciles a VirtualMachine object
+type VirtualMachineReconciler struct {
```

Comment:
I am not sure we need to separate Reconcilers, I think we can make one for both VM and VMIs and just check the expected condition `shouldCleanFinalizers` and remove the finalizer is so.

Reply:
I prefer please the current separation, easier to maintain and less mental burden imo
each controller does its own,
the VM handles VM events, and VMI handles VMI events, also each one is self contained
and handles it own cases